modules/nf-core/samtools/sormadup/main.nf

```diff
@@ -0,0 +1,99 @@
+process SAMTOOLS_SORMADUP {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.19.2--h50ea8bc_0' :
+        'biocontainers/samtools:1.19.2--h50ea8bc_0' }"
+
+    input:
+    tuple val(meta), path(input)
+    tuple val(meta2), path(fasta)
+
+    output:
+    tuple val(meta), path("*.{bam,cram}")   , emit: markdup
+    tuple val(meta), path("*.{bai,crai}")   , emit: markdup_index, optional: true
+    tuple val(meta), path("*.metrics")      , emit: metrics
+    path "versions.yml"                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args ?: ''
+    def args3 = task.ext.args ?: ''
+    def args4 = task.ext.args ?: ''
+    def args5 = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def extension = args.contains("--output-fmt sam") ? "sam" :
+                    args.contains("--output-fmt bam") ? "bam" :
+                    args.contains("--output-fmt cram") ? "cram" :
+                    "bam"
+    def reference = fasta ? "--reference ${fasta}" : ""
+    def sort_memory = (task.memory.mega/task.cpus*0.75).intValue()
+
+    """
+    samtools cat \\
+        --threads $task.cpus \\
+        $args2 \\
+        ${input}  \\
+    | \\
+    samtools collate \\
+        $args3 \\
+        -O \\
+        -u \\
+        -T ${prefix} \\
+        --threads $task.cpus \\
+        ${reference} \\
+        - \\
+    | \\
+    samtools fixmate \\
+        $args4 \\
+        -m \\
+        -u \\
+        --threads $task.cpus \\
+        - \\
+        - \\
+    | \\
+    samtools sort \\
+        $args5 \\
+        -u \\
+        -T ${prefix} \\
+        --threads $task.cpus \\
+        -m ${sort_memory}M \\
+        - \\
+    | \\
+    samtools markdup \\
+        -T ${prefix} \\
+        -f ${prefix}.metrics \\
+        --threads $task.cpus \\
+        $args \\
```

Comment:
> The numbering of each $args variable MUST correspond to the order of the tools, and MUST be documented in meta.yml. E.g. in the first example, bwa mem is the first tool so is given $args, samtools view is the second tool so is $args2, etc.

https://nf-co.re/docs/contributing/modules#new-module-guidelines-and-pr-review-checklist

https://nf-co.re/docs/contributing/modules#documentation

Reply:
Is there an example somewhere of adding `args` to meta.yml? I want to do it right from the get go.

Reply:
Can search modules for `args_id` in the `meta.yml`s for concrete examples. I'm referring to point 5 in the documentation link I posted above.

Reply:
Fixed it! Can you check again?

Reply:
LGTM